src/lib/Services/error-message.service.ts

```diff
@@ -7,12 +7,14 @@ import { CUSTOM_ERROR_MESSAGES } from "../Tokens/tokens";
 export class ErrorMessageService {
   private defaultErrors = DEFAULT_ERRORS;
 
-  constructor(
-    @Inject(CUSTOM_ERROR_MESSAGES)
-    public customErrorMessages: ErrorMessage[] = []
-  ) {}
+  public errorMessages: ErrorMessage[];
 
-  get errorMessages() {
-    return [...this.customErrorMessages, ...this.defaultErrors];
+  constructor(
+    @Inject(CUSTOM_ERROR_MESSAGES) public customErrorMessages: ErrorMessage[][]
+  ) {
+    this.errorMessages = customErrorMessages.reduce(
```

Comment:
What's the reasoning for using `reduce` instead of just spreading?

Reply:
I think I see. `customErrorMessages` is an array of arrays.

Reply:
Since we don't know how many arrays of arrays we might encounter, I used reduce.